nexus/src/db/lookup.rs

```diff
@@ -0,0 +1,723 @@
+                        datastore,
+                        name
+                    ).await?;
+                opctx.authorize(authz::Action::Read, &authz_child).await?;
+                Ok((authz_child, db_child))
+            }
+
+            impl LookupNoauthz for $pc<'_> {
+                type LookupType = authz::$pc;
+
+                fn lookup(
+                    &self,
+                ) -> BoxFuture<'_, LookupResult<Self::LookupType>> {
+                    async {
+                        let lookup = self.lookup_root();
+                        let opctx = &lookup.opctx;
+                        let datastore = lookup.datastore;
+                        match self.key {
+                            Key::Name(_, name) => {
+                                let (rv, _) =
+                                    [<$pc:lower _lookup_by_name_no_authz>](
+                                        opctx,
+                                        datastore,
+                                        name
+                                    ).await?;
+                                Ok(rv)
+                            }
+                            Key::Id(_, id) => {
+                                let (rv, _) =
+                                    [<$pc:lower _lookup_by_id_no_authz>](
+                                        opctx,
+                                        datastore,
+                                        id
+                                    ).await?;
+                                Ok(rv)
+                            }
+                        }
+                    }.boxed()
+                }
+            }
+
+            impl Fetch for $pc<'_> {
+                type FetchType = (authz::$pc, model::$pc);
+
+                fn fetch(&self) -> BoxFuture<'_, LookupResult<Self::FetchType>> {
+                    async {
+                        let lookup = self.lookup_root();
+                        let opctx = &lookup.opctx;
+                        let datastore = lookup.datastore;
+                        match self.key {
+                            Key::Name(_, name) => {
+                                [<$pc:lower _fetch_by_name>](
+                                    opctx,
+                                    datastore,
+                                    name
+                                ).await
+                            }
+                            Key::Id(_, id) => {
+                                [<$pc:lower _fetch_by_id>](
+                                    opctx,
+                                    datastore,
+                                    id
+                                ).await
+                            }
+                        }
+                    }
+                    .boxed()
+                }
+            }
+        }
+    };
+}
+
+macro_rules! define_lookup_with_parent {
+    (
+        $pc:ident,              // Pascal-case version of resource name
```

Comment:
Is "pc" supposed to stand for "pascal-case"?

Could we just call this "resource"? Identifying a variable exclusively by the naming convention is IMO a little obscure

Reply:
Agreed -- but this code is all gone anyway.  Resolving this thread.